### ISSUE TITLE
Display definition requirements

### ISSUE THREAD

**Issue by byorgey** [swarm-game/swarm#922]
> What I see as the actual problem is that the requirements are hidden. We should be able to show them...

_Originally posted by @xsebek in https://github.com/swarm-game/swarm/issues/921#issuecomment-1356190044_

Conceptually, the requirements of a command are part of its type (eventually, with #231, they will be literally so), and it should be something a player can see about a definition, just as it is currently possible to display the type of a definition.

Since the set of requirements can be quite large, however, displaying them should probably not be the default.  But perhaps there could be a key to toggle whether requirements are shown as part of types.  Or perhaps there could be some sort of built-in command to display the requirements of a name.

**Comment by byorgey** [comment]
I'm currently imagining a built-in primitive `requirements` which takes a term as an argument and generates the list of requirements as a message in your log.  So for example if you had defined
```
def f = require "boat"; require 5 "rock"; move end
```
then typing `requirements f` at the REPL will generate a log message that looks like this:
```
Requirements for `f`:
  Equipment:
    - treads
    - boat
  Inventory:
    - rock (5)
```

**Comment by byorgey** [comment]
Putting it in the REPL makes sense, but it is not a value so we would have to add some kind of special case for displaying it at the REPL.  More importantly, I am worried it will be too long to fit, and until we implement #60 anything that scrolls off the top of the REPL is gone forever.

**Comment by byorgey** [comment]
For posterity, I am adding a note here to say that I have implemented this in #1183 as a message in the log.  In practice, it is hard for me to imagine a scenario where you would have a `3D printer` and thus care about the requirements for doing a `build`, but not have a `logger`.  A `logger` is very easy to make, and it's standard equipment for most scenarios.
